--- SDL-1.2.15.org/src/video/fbcon/SDL_fbvideo.c	2012-01-19 07:30:06.000000000 +0100
+++ SDL-1.2.15/src/video/fbcon/SDL_fbvideo.c	2012-02-04 23:34:04.000000000 +0100
@@ -34,6 +34,7 @@
 #include <asm/page.h>		/* For definition of PAGE_SIZE */
 #endif
 
+#include <linux/fb.h>
 #include <linux/vt.h>
 
 #include "SDL_video.h"
@@ -1409,8 +1410,9 @@
 
 static void FB_WaitVBL(_THIS)
 {
-#ifdef FBIOWAITRETRACE /* Heheh, this didn't make it into the main kernel */
-	ioctl(console_fd, FBIOWAITRETRACE, 0);
+#ifdef FBIO_WAITFORVSYNC
+	unsigned int screen = 0;
+	ioctl(console_fd, FBIO_WAITFORVSYNC, &screen);
 #endif
 	return;
 }
