--- a/Alc/ALu.c
+++ b/Alc/ALu.c
@@ -799,15 +799,7 @@
 
     SuspendContext(ALContext);
 
-#if defined(HAVE_FESETROUND)
-    fpuState = fegetround();
-    fesetround(FE_TOWARDZERO);
-#elif defined(HAVE__CONTROLFP)
-    fpuState = _controlfp(0, 0);
-    _controlfp(_RC_CHOP, _MCW_RC);
-#else
     (void)fpuState;
-#endif
 
     //Figure output format variables
     BlockAlign  = aluChannelsFromFormat(format);
